controllers/warehousesController.js

```diff
@@ -0,0 +1,108 @@
+const Joi = require("joi");
+const knex = require("knex")(require("../knexfile"));
+
+// {
+//     "warehouse_id": 1,
+//     "item_name": "Paper Towels",
+//     "description": "Made out of military-grade synthetic materials, these paper towels are highly flammable, yet water resistant, and easy to clean.",
+//     "category": "Gear",
+//     "status": "Out of Stock",
+//     "quantity": 0
+//   }
+
+// POST -- Create New InveNtory of Warehouse by its ID
+const createNewWarehouse = async (req, res) => {
+  try {
+    // inventorySchema for req body validations
+    const warehouseSchema = Joi.object({
+      warehouse_name: Joi.string().required(),
+      address: Joi.string().required(),
+      contact_name: Joi.string().required(),
+      contact_information: Joi.string().required(),
+      actions: Joi.string().required().required(),
+    });
+
```

Comment:
Kindly look at the Ticket SPF2-16  and verify the data types that will be sent to the req.body, you are missing some values and have added some values that should not be there

Reply:
thank you! updated now